common/failingset.go

```diff
@@ -99,10 +103,18 @@ func (i *FailingSet) observe(_ context.Context, result metric.Int64ObserverResul
 		for name := range nm {
 			names = append(names, name)
 		}
+		if names == nil {
```

Comment:
I wanted to check if `names` is `nil` since my IDE is complaining that indexing could panic if I didn't check if the slice is `nil` first.  

But that actually can't happen, since the indexing `names = names[:failingMetricMaxReportedMetrics]` is scoped inside the size check, `count > failingMetricMaxReportedMetrics`. Guess the IDE is not that smart after all.